packages/querygrid/src/components/user/actions.ts

```diff
@@ -0,0 +1,107 @@
+import moment from 'moment'
+import { Map, OrderedMap } from 'immutable'
+import { Ajax , ActionURL, Utils } from '@labkey/api'
+import { caseInsensitive, hasAllPermissions, PermissionTypes, SchemaQuery, User, buildURL } from "@glass/base";
+import { ChangePasswordModel } from "./models";
+
+export function getUserPermissionsDisplay(user: User): Array<string> {
+    let permissions = [];
+
+    if (user.isAdmin) {
+        permissions.push('Administrator');
+    }
+    else {
+        if (hasAllPermissions(user, [PermissionTypes.DesignSampleSet])) {
+            permissions.push('Sample Set Designer');
+        }
+        if (hasAllPermissions(user, [PermissionTypes.DesignAssay])) {
+            permissions.push('Assay Designer');
+        }
+        permissions.push(user.canUpdate ? 'Editor' : (user.canInsert ? 'Author' : 'Reader'));
+    }
+
+    return permissions
+}
+
+export function getUserLastLogin(userProperties: Map<string, any>, dateFormat: string): string {
+    const lastLogin = caseInsensitive(userProperties.toObject(), 'lastlogin');
+    return lastLogin ? moment(lastLogin).format(dateFormat) : undefined;
+}
+
+export function getUserDetailsRowData(user: User, data: OrderedMap<string, any>, avatar: File): FormData {
+    const formData = new FormData();
+    const row = {UserId: user.id, ...data.toJS()};
+
+    Object.keys(row).forEach((key) => {
+        let value = row[key];
+
+        // need to convert booleans to string for save
+        if (value !== null && Utils.isBoolean(value)) {
+            value = value.toString();
+        }
+
+        // need to remove email from the posted data since that is not an updatable value for this action
+        if (key.toLowerCase() === 'email') {
+            value = undefined;
+        }
+
+        if (value !== undefined) {
+            formData.append(key, value);
```

Comment:
Doesn't this mean they can't remove any values that have previously been set?

Reply:
Nope, removing values does work. Removed values come through as empty strings I believe.